SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -149,14 +168,17 @@ void CFlowOutput::SetAnalyzeSurface(const CSolver* const*solver, const CGeometry
   const bool incompressible = config->GetKind_Regime() == ENUM_REGIME::INCOMPRESSIBLE;
   const bool energy         = config->GetEnergy_Equation();
   const bool streamwisePeriodic = (config->GetKind_Streamwise_Periodic() != ENUM_STREAMWISE_PERIODIC::NONE);
-  const bool species        = config->GetKind_Species_Model() != SPECIES_MODEL::NONE;
+  const bool species        = config->GetKind_Species_Model() == SPECIES_MODEL::SPECIES_TRANSPORT;
+  const bool flamelet       = config->GetKind_Species_Model() == SPECIES_MODEL::FLAMELET;
   const auto nSpecies       = config->GetnSpecies();
+  const auto nScalars       = config->GetNScalars();
 
   const bool axisymmetric               = config->GetAxisymmetric();
   const unsigned short nMarker_Analyze  = config->GetnMarker_Analyze();
 
   const auto flow_nodes = solver[FLOW_SOL]->GetNodes();
   const CVariable* species_nodes = species ? solver[SPECIES_SOL]->GetNodes() : nullptr;
+  const CVariable* scalar_nodes  = flamelet ? solver[SPECIES_SOL]->GetNodes() : nullptr;
```

Comment:
We cannot have species and flamelet at the same time due to the shared solver right?
In which case please use `species || flamelet` and that way the code duplication is saved.

Reply:
Exactly, it's either species or flamelet. I noticed that scalar_nodes is not used anymore because we replaced it with custom output so I removed this line. We can do the same thing for the species transport, but let's do that in another PR.